src/setuptools_scm/config.py

```
@@ -120,4 +119,5 @@ def from_file(cls, name="pyproject.toml"):
         """
         with open(name) as strm:
             defn = __import__("toml").load(strm)
-        return cls()._load(defn.get("tool", {})["setuptools_scm"])
+        section = defn.get("tool", {})["setuptools_scm"]
+        return cls(**section)
```

The danger (and benefit) to this approach is it couples the class interface to the configuration schema. It's now invalid to have any values in the configuration that aren't supported by the class, which means now users must be much more sensitive to the version of setuptools_scm used, making it more difficult to be forward-compatible (supplying a value that is used if supported but ignored otherwise). It was my intention to be suitably lenient, acknowledging the risk that a user might supply the _wrong_ key (with a typo or something) and that typo would fail silently.

If you want to go with this approach, I'd recommend allowing `**kwargs` to `Configuration()` and (if desired) warn if unexpected keyword arguments are found.

---

good point, i#ll think about how to layer it in correctly